consult.el

```diff
@@ -1357,22 +1357,36 @@ See `isearch-open-necessary-overlays' and `isearch-open-overlay-temporary'."
                 restore))))
     restore))
 
-(defun consult--jump-1 (pos)
-  "Go to POS and recenter."
-  (if (and (markerp pos) (not (marker-buffer pos)))
-      ;; Only print a message, no error in order to not mess
-      ;; with the minibuffer update hook.
-      (message "Buffer is dead")
-    ;; Switch to buffer if it is not visible
-    (when (and (markerp pos) (not (eq (current-buffer) (marker-buffer pos))))
-      (consult--buffer-action (marker-buffer pos) 'norecord))
-    ;; Widen if we cannot jump to the position (idea from flycheck-jump-to-error)
-    (unless (= (goto-char pos) (point))
-      (widen)
-      (goto-char pos))))
+(defmacro consult--with-jump (pos &rest body)
+  "Go to POS."
+  (declare (indent 1))
+  `(if (and (markerp ,pos) (not (marker-buffer ,pos)))
+       ;; Only print a message, no error in order to not mess
+       ;; with the minibuffer update hook.
+       (message "Buffer is dead")
+     (let* ((goto-other (and (markerp ,pos) (not (eq (current-buffer) (marker-buffer ,pos)))))
+            (w (cond (goto-other
+                      (consult--buffer-action (marker-buffer ,pos) 'norecord)
+                      (get-buffer-window (marker-buffer ,pos)))
+                     (t
+                      (get-buffer-window (current-buffer)))))
+            (f (lambda ()
+                 (unless (= (goto-char ,pos) (point))
+                   (widen) ;; Widen if we cannot jump to the position (idea from flycheck-jump-to-error)
+                   (goto-char ,pos))
+                 ,@body
+                 (run-hooks 'consult-after-jump-hook))))
+       (if (window-live-p w)
+           (with-selected-window w (funcall f))
```

Comment:
Here is taking care of the case like

* `current-buffer` is not the target buffer
* the window of the target buffer is not selected window